cpu/stm32_common/periph/gpio.c

```diff
@@ -87,6 +87,8 @@ int gpio_init(gpio_t pin, gpio_mode_t mode)
     periph_clk_en(AHB1, (RCC_AHB1ENR_GPIOAEN << _port_num(pin)));
 #endif
 
+    /* do not touch actual line state */
+    _port(pin)->ODR = _port(pin)->IDR;
```

Comment:
I'm really not sure about it but shouldn't the initial state be dependent on the `mode`? In that case you would only set `ODR=IDR` for the open drain case.

Reply:
@niziak are you still around? Can you give a quick feedback?

Reply:
> I'm really not sure about it but shouldn't the initial state be dependent on the `mode`? In that case you would only set `ODR=IDR` for the open drain case.

I think you are right. I'll prepare change.

Reply:
Ok just ping me when you're done

Reply:
@niziak still around?